trackers/core/ksptracker/tracker.py

```diff
@@ -0,0 +1,252 @@
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+import networkx as nx
+import numpy as np
+import supervision as sv
+
+from trackers.core.base import BaseTracker
+
+
+@dataclass(frozen=True)
+class TrackNode:
+    """
+    Represents a detection node in the tracking graph.
+
+    Attributes:
+        frame_id (int): Frame index where detection occurred
+        grid_cell_id (int): Discretized grid cell ID of detection center
+        position (tuple): Grid coordinates (x_bin, y_bin)
+        confidence (float): Detection confidence score
+    """
+
+    frame_id: int
+    grid_cell_id: int
+    position: tuple
+    confidence: float
+
+    def __hash__(self) -> int:
+        """
+        Generate hash using frame and grid cell.
+
+        Returns:
+            int: Hash value for node
+        """
+        return hash((self.frame_id, self.grid_cell_id))
+
+    def __eq__(self, other: Any) -> bool:
+        """
+        Compare nodes by frame and grid cell ID.
+
+        Args:
+            other (Any): Object to compare
+
+        Returns:
+            bool: True if same node, False otherwise
+        """
+        if not isinstance(other, TrackNode):
+            return False
+        return (self.frame_id, self.grid_cell_id) == (
+            other.frame_id,
+            other.grid_cell_id,
+        )
+
+
+class KSPTracker(BaseTracker):
+    """
+    Offline tracker using K-Shortest Paths (KSP).
+
+    Attributes:
+        grid_size (int): Size of each grid cell (in pixels)
+        max_gap (int): Max frame gap between connections
+        min_confidence (float): Minimum detection confidence
+        max_distance (float): Max dissimilarity (1 - IoU) allowed
+    """
+
+    def __init__(
+        self,
+        grid_size: int = 20,
+        max_gap: int = 30,
+        min_confidence: float = 0.3,
+        max_distance: float = 0.3,
+    ) -> None:
+        """
+        Initialize KSP tracker with config parameters.
+
+        Args:
+            grid_size (int): Pixel size of each grid cell
+            max_gap (int): Max frames between connected detections
+            min_confidence (float): Min detection confidence
+            max_distance (float): Max allowed dissimilarity
+        """
+        self.grid_size = grid_size
+        self.max_gap = max_gap
```

Comment:
`self.max_gap` doesn't seem to be used in this implementation. According to the paper, a key strength of KSP Tracker is the ability to bridge gaps caused by false negatives (missed detections).